samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/SharedTestUtils.kt

```diff
@@ -36,24 +64,30 @@ internal fun createMockWebServer(): MockWebServer {
  */
 fun waitForPingContent(
     pingName: String,
+    pingReason: String?,
     server: MockWebServer,
     maxAttempts: Int = 3
 ): JSONObject?
 {
+    var parsedPayload: JSONObject? = null
     var attempts = 0
     do {
         attempts += 1
-        val request = server.takeRequest(20L, TimeUnit.SECONDS)
+        val request = server.takeRequest(20L, TimeUnit.SECONDS) ?: continue
         val docType = request.path.split("/")[3]
         if (pingName == docType) {
-            return if (request.getHeader("Content-Encoding") == "gzip") {
-                val bodyInBytes = ByteArrayInputStream(request.body.readByteArray()).readBytes()
-                JSONObject(decompressGZIP(bodyInBytes))
-            } else {
-                JSONObject(request.body.readUtf8())
+            parsedPayload = JSONObject(request.getPlainBody())
+            if (pingReason == null) {
+                break
```

Comment:
I changed this because ktlint complains about the three returns in the same function.

Reply:
You can also just suppress the warning if the code looks reasonably clearer the other way.